backend/src/resolvers.js

```diff
@@ -0,0 +1,73 @@
+const { UserInputError } = require("apollo-server");
+
+const resolvers = {
+    Query: {
+        users: (parent, args, context) => context.dataSources.db.allUsers(),
+        posts: (parent, args, context) => context.dataSources.db.allPosts(),
+    },
+    Mutation: {
+        write: async (parent, args, context) => {
+            const newPost = {
+                title: args.post.title,
+                authorName: args.post.author.name,
+            };
```

Comment:
Personally, these resolvers contain too much logic to construct response objects, but that's a matter of taste I reckon.
Good code style in any case!